src/main/java/org/isf/medicalstock/manager/MovStockInsertingManager.java

```diff
@@ -114,58 +115,113 @@ protected void validateMovement(Movement movement, boolean checkReference) throw
+			if (isCharge && isAutomaticLotIn()) {
+				// check everything but the code
+				validateLot(errors, lot, false);
 			}
 
-			if (lot.getPreparationDate() == null) {
-				errors.add(new OHExceptionMessage(MessageBundle.getMessage("angal.medicalstock.insertavalidpreparationdate.msg")));
-			}
-
-			if (lot.getPreparationDate() != null && lot.getDueDate() != null && lot.getPreparationDate().isAfter(lot.getDueDate())) {
-				errors.add(new OHExceptionMessage(MessageBundle.getMessage("angal.medicalstock.thepreparationdatecannotbyaftertheduedate.msg")));
-			}
+			/*
+			 * Check Lot code: it must be unique per Lot-Medical
+			 * 
+			 */
 			List<Integer> medicalIds = ioOperations.getMedicalsFromLot(lot.getCode());
 			if (movement.getMedical() != null && !(medicalIds.isEmpty() || medicalIds.size() == 1 && medicalIds.get(0).intValue() == movement
 							.getMedical().getCode().intValue())) {
 				errors.add(new OHExceptionMessage(MessageBundle.getMessage("angal.medicalstock.thislotreferstoanothermedical.msg")));
 			}
-			if (GeneralData.LOTWITHCOST && chargingType) {
+
+			/*
+			 * Check cost.
+			 * 
+			 * - charge:
+			 * 
+			 * LOTWITHCOSTS=yes, Lot(..., cost) with cost > 0.0, otherwise error
+			 * 
+			 * LOTWITHCOSTS=no, no check: with or without cost, the cost will be ignored
+			 * 
+			 */
+			if (isCharge && GeneralData.LOTWITHCOST) {
 				BigDecimal cost = lot.getCost();
 				if (cost == null || cost.doubleValue() <= 0.0) {
 					errors.add(new OHExceptionMessage(MessageBundle.getMessage("angal.medicalstock.multiplecharging.zerocostsarenotallowed.msg")));
 				}
 			}
+
+			/*
+			 * Check quantity.
+			 * 
+			 * AUTOMATICLOT_OUT=no, specified quantity must be equal or lower than the lot quantity
+			 * 
+			 * AUTOMATICLOT_OUT=yes, no check: the quantity will be split automatically between available lots
+			 */
 			if (!isAutomaticLotOut()) {
 
-				if (movement.getType() != null && !chargingType && movement.getQuantity() > lot.getMainStoreQuantity()) {
+				if (movement.getType() != null && !isCharge && movement.getQuantity() > lot.getMainStoreQuantity()) {
 					errors.add(new OHExceptionMessage(MessageBundle.getMessage("angal.medicalstock.movementquantityisgreaterthanthequantityof.msg")));
 				}
 			}
-		} else {
-			if (!chargingType && !isAutomaticLotOut()) {
 
-				errors.add(new OHExceptionMessage(MessageBundle.getMessage("angal.medicalstock.choosealot.msg")));
-			}
+		} else {
+			errors.add(new OHExceptionMessage(MessageBundle.getMessage("angal.medicalstock.choosealot.msg")));
 		}
 
 		if (!errors.isEmpty()) {
 			throw new OHDataValidationException(errors);
 		}
 	}
 
+	private void validateLot(List<OHExceptionMessage> errors, Lot lot, boolean checkCode) {
+		if (checkCode) {
+			if (lot.getCode().length() >= 50) {
+				errors.add(new OHExceptionMessage(MessageBundle.getMessage("angal.common.error.title"),
```

Comment:
why are you using the long version of this method?  The single parameter version sets the title and serverity.

It could be:

`errors.add(new OHExceptionMessage(MessageBundle.getMessage("angal.medicalstock.thelotidistoolongmax50chars.msg")));`

See line 156 for example.

Reply:
yes, sorry, I was copying from an old code where `validateLot` method was still separated. Fixed. Thanks!